packages/core/src/material/PBRMaterial.ts

```diff
@@ -299,19 +309,43 @@ export class PBRMaterial extends PBRBaseMaterial {
 
   /**
    * Refraction switch.
+   * @remarks Use refractionMode to set the refraction shape.
    */
-  get enableRefraction(): boolean {
-    return this._refractionEnabled;
+  get refractionMode(): RefractionMode {
+    return this._refractionMode;
   }
 
-  set enableRefraction(value: boolean) {
-    if (value !== this._refractionEnabled) {
-      this._refractionEnabled = value;
+  set refractionMode(value: RefractionMode) {
+    if (value !== this._refractionMode) {
       if (value) {
-        this.shaderData.enableMacro(PBRMaterial._refactionMacro);
+        this.shaderData.enableMacro(PBRMaterial._refractionMacro);
+        this.renderState.renderQueueType = RenderQueueType.Transparent;
+        this.renderState.blendState.targetBlendState.enabled = false;
       } else {
-        this.shaderData.disableMacro(PBRMaterial._refactionMacro);
+        this.shaderData.disableMacro(PBRMaterial._refractionMacro);
+        this.renderState.renderQueueType = RenderQueueType.Opaque;
+        this.renderState.blendState.targetBlendState.enabled = true;
```

Comment:
nonono, you should consider the value previously set by the user，such as `isTransparent` 、`renderFace` 、`alphaCutoff` 、`blendMode`

Reply:
you can try these relation at Unity:
![image](https://github.com/user-attachments/assets/ae5a3b50-621c-4bd6-8403-acb88e41a04b)